app/src/main/java/org/fossasia/openevent/api/APIClient.java

```diff
@@ -37,6 +48,15 @@ public APIClient() {
                 .setConverter(new GsonConverter(gson))
                 .setEndpoint(Urls.BASE_GET_URL_ALT)
                 .setLogLevel(RestAdapter.LogLevel.FULL)
+                .setErrorHandler(new ErrorHandler() {
+                    @Override
+                    public Throwable handleError(RetrofitError cause) {
+                        ErrorDialogEvent errEvent = new ErrorDialogEvent(null, null);
+                        errEvent.ErrorHandlerFunction(cause, errEvent);
+                        OpenEventApp.postEventOnUIThread(new ErrorDialogEvent(errEvent.ErrorType, errEvent.ErrorDesc));
+                        return new Exception(cause);
```

Comment:
if the constructor is available you can just create the object right? Why do you need to initialise it with `null` and then pass it. The thing I don't get is that whats the point of having a member function when you are using it as a static helper method. Plus your datastructure returns nothing and you create another copy of the object from it, which is completely baffling. Objects are *NOT* cheap.